packages/open-ui/src/components/Button/Button.tsx

```diff
@@ -2,26 +2,26 @@ import React from 'react';
 import classNames from 'classnames';
 
 import { MarginStyleKeys } from 'interfaces/MarginStyleKeys';
-import { StyleTypeOption } from 'interfaces/StyleTypeOption';
+import { IntentOption } from 'interfaces/IntentOption';
 import { SizeOption } from 'interfaces/SizeOption';
 import { getMarginProps } from 'utils/helpers';
+import { toSentenceCase } from 'utils/strings/to-sentence-case';
 import styles from './Button.module.scss';
 
 interface ButtonProps {
-  type?: StyleTypeOption;
+  intent?: IntentOption;
+  onClick: () => void;
```

Comment:
@remotealex lets add a proper type here to catch the event `onClick(e: React.MouseEvent<HTMLButtonElement>) => void`

Reply:
We're not actually passing the event back to them so this is the correct type

Reply:
you mean like we are passing this onClick handler to the users but actual event is not generated here ryt ?

Reply:
```      
    onClick={e => {
      e.preventDefault();
      e.stopPropagation();
      e.currentTarget.blur();
      onClick();
      return null;
    }}
```
See how we dont pass `e` to their function

Reply:
yeah exactly was mentioning the same